providers/mysql_basic.rb

```
@@ -7,82 +7,46 @@
+  bash 'wait_mysqld_started' do
     user "root"
+    retries num_retry
+    retry_delay ret_delay 
     code <<-EOF
-    set -e || set -o pipefail
-    service mysqld restart
-    sleep 5
-    if [ `#{node['mysql']['base_dir']}/bin/mysqladmin -u root -S #{node['ndb']['mysql_socket']} status` -ne 0 ] ; then
-     wait=new_resource.wait_time
-     timeout=0
-     while [ $timeout -lt $wait ] ; do
-         echo -n "."
-         sleep 1
-         if [ `#{node['mysql']['base_dir']}/bin/mysqladmin -u root -S #{node['ndb']['mysql_socket']} status` -eq 0 ] ; then
-           timeout=new_resource.wait_time
-         fi
-         timeout=`expr $timeout + 1`
-     done
-      # If it did't work, try starting it again...
-        if [ `#{node['mysql']['base_dir']}/bin/mysqladmin -u root -S #{node['ndb']['mysql_socket']} status` -ne 0 ] ; then
-          service mysqld start
-          sleep new_resource.wait_time
-        fi
-    fi
-
-    # The mysqld may really not have started...
-    if [ `#{node['mysql']['base_dir']}/bin/mysqladmin -u root -S #{node['ndb']['mysql_socket']} status` -ne 0 ] ; then
-       echo "Something went badly wrong. Couldn't start mysqld. Trying to reinstall. Backing up to #{node['ndb']['base_dir']}/backup_mysql/"
-       mkdir -p #{node['ndb']['base_dir']}/backup_mysql
-       mv #{node['ndb']['mysql_server_dir']}* #{node['ndb']['base_dir']}/backup_mysql/
-
-       export MYSQL_HOME=#{node['ndb']['root_dir']}
-       cd #{node['mysql']['base_dir']}
-       su #{node['ndb']['user']} -c "./bin/mysqld --defaults-file=#{node['ndb']['root_dir']}/my.cnf --initialize-insecure --explicit_defaults_for_timestamp"
-
-       service mysqld start
-       sleep new_resource.wait_time
-    fi
-
-    if [ `#{node['mysql']['base_dir']}/bin/mysqladmin -u root -S #{node['ndb']['mysql_socket']} status` -ne 0 ] ; then
-         exit 1
-    fi
+      #{node['mysql']['base_dir']}/bin/mysqladmin -u root -S #{node['ndb']['mysql_socket']} status
```

wow, big delete here. i can't say i agree with it. Those checks were there for a reason (at the time) - to make the check more robust. Sometimes, mysqld should wait until ndbmtds come up, not just fail early - which is terrible for karamel.

---

It's more a rewrite than a delete - I just transformed the bash code in a more "chefy" style. The status command will be executed `retries` times and wait 5 seconds in between retries. Just as the bash code was doing. 

The only thing I removed was the silent re-initialization of the mysql server if it didn't start. This is not what I would expect as admin. If something goes bad and a re-initialization is required I need to know. During a re-initialization the views will be lost for instance. Iif you don't know that the MySQL server has been re-initialized, you will spend quite some time trying to figure out why you are not able to log in into Hopsworks (experience talking here :D)

---

Case in point, Today I was updating Gautier's instance. MySQL took some time to come, so the recipe re-initialized it and the views were gone.